Bug 1768142 - Create a checklist for helping with data validation. (#2047)
mozilla/glean: docs/user/user/metrics/validation-checklist.md

# Validating the collected data

> Once instrumentation is added to a product, it is worth investing time early to understand if the data looks as expected and reasonable, while clients start sending it, and act if it does not.

Reviewer: Can we reword this a bit to remove some of the commas?

```suggestion
It is worth investing time when instrumentation is added to the product to understand if the data looks reasonable and expected, and to take action if it does not.
```

The following checklist could help guide such validation effort.

> 1. Before releasing the product with the new data collection, make sure the data looks as expected by generating sample data on a local machine and inspecting it on the [Glean Debug View](../../reference/debug/debugViewTag.md):

Reviewer 2: Should we link to [the debugging overview](https://mozilla.github.io/glean/book/reference/debug/index.html) instead? (( That reminds me, I do need to get to https://bugzilla.mozilla.org/show_bug.cgi?id=1762281 sooner rather than later ))

    a. is the data showing up in the correct ping(s)?

    b. does the metric report the expected data?

    c. if exercising the same path again, it is expected for the data to be submitted again? And does it?

> 2. As users start adopting the version of the product with the new data collection, the initial data coming in should be checked, to understand how the measurements are behaving in the wild:

Reviewer 2: Give an example duration? For Mozilla projects, they're all subject to the ["about two days"](https://blog.mozilla.org/data/2021/12/14/this-week-in-glean-how-long-must-i-wait-before-i-can-see-my-data/) estimate due to pipeline effects. Even a rough estimate "As users start adopting the version of the product with the new data collection (usually within a few days of release), ..." might help guide folks' expectations.

>     a. does the shape of the collected data fit our prior expectations?

Reviewer 2: Which "our"?

Is this point "Does this organcially-sent data satisfy they same quality expectations the manually-sent data did in Step 1"?

>     b. is there any new [error](../../user/metrics/error-reporting.md) being reported for the new data points? If so, does this point to an edge case that should be documented and/or fixed in the code?

Reviewer 2: s/is/are/

Reviewer 2: Oh, and this is a good opportunity to highlight that you _could_ fix it in the code, but you could also [fix it live](https://github.com/mozilla/glean-annotations/)

Reviewer 2: Oh oh, and there's a good opportunity here to encourage folks to check the Glean Dictionary to ensure their instrumentation's information is showing up properly. Not that I expect that to ever be a problem, but it might be nice to remind folks that the dictionary exists.
